src/audio/mux.c

```diff
@@ -59,12 +97,94 @@ static int mux_params(struct comp_dev *dev)
 	return 0;
 }
 
+/**
+ * \brief Sets mux control command.
+ * \param[in,out] dev mux base component device.
+ * \param[in,out] cdata Control command data.
+ * \return Error code.
+ */
+static int mux_ctrl_set_cmd(struct comp_dev *dev,
+			       struct sof_ipc_ctrl_data *cdata)
+{
+	struct mux_data *cd = comp_get_drvdata(dev);
+	int j;
+
+	/* validate */
+	if (cdata->num_elems == 0 || cdata->num_elems > SOF_IPC_MAX_CHANNELS) {
+		trace_mux_error("mux_ctrl_set_cmd() error: "
+				   "invalid cdata->num_elems");
+		return -EINVAL;
+	}
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_ENUM:
+		/*FIXME: fix logger for keywork */
+		trace_mux("mux_ctrl_set_cmd(), SOF_CTRL_CMD_ENUM, ");
+
+		/* save switch state */
+		for (j = 0; j < cdata->num_elems; j++)
```

Comment:
what exactly is stored in num_elements when for cmd == SOF_CMD_ENUM ? is it a number of parameters and each chanv[j].value represents enum value of parameter[0..num_elements] ?

Reply:
its the number of channel elements to be changed.